### DISCUSSION TITLE
GitHub Classroom Pricing

### DISCUSSION THREAD

**Discussion by BrookesDavidS** [community/community#123676] [GitHub Education]
### Hi everyone,

I am considering the use of GitHub Classroom for the delivery of a course, but I am finding it difficult to work out what it might cost my institution. The course will have around 200 students, each student is expected to complete around 10 small assignments each week. I am not proposing to use CodeSpaces.

Rather more specifically, the questions I have are:

  1. Are students on a classroom supposed to be outside collaborators of the organization that owns the classroom (as opposed to members)?
  2. If students are outside collaborators, am I correct in thinking that action minutes, storage, and other usage incurred by a student will be deducted from the allowance for the student's own GitHub account, rather than that of the organization that owns the classroom?
  3. Would it be feasible to run a course of the nature outlined above without using any paid-for accounts (either for the organization or the students)?

**Comment by abhiperfect** [comment]
🎓 **GitHub Classroom Cost Considerations**:

1. **Student Access**: Students are collaborators in the organization that owns the classroom, granting them access to course repositories.

2. **Usage Costs**: Actions by students count towards the organization's usage limits, ensuring smooth operation within plan allowances.

3. **Feasibility Without Paid Accounts**: It's feasible to run the course without paid accounts by carefully monitoring usage and utilizing GitHub's free plans.

4. **Optimizing Usage**: Encourage efficient coding practices to manage repository size and usage effectively.

5. **Upgrade Options**: Consider upgrading to a paid plan if usage exceeds free allowances or specific features are needed.

By leveraging GitHub Classroom effectively and managing usage, delivering the course seamlessly within budgetary constraints becomes achievable and rewarding for all involved.

**Reply by BrookesDavidS** [reply]
@abhiperfect Thanks very much for that prompt and detailed response. On the basis of what you have said, I doubt that I would be able to run my course using free accounts (it might work with more experienced students, but I am teaching beginners).

If I am to get approval for GitHub classroom I am going to have to calculate an upper bound on the cost that the organization, and the students, might incur when using paid-for accounts. Does anyone have any hints on how to do that?

Just for clarity, I am aware that I could go into the billing page of an account and set a spending limit. But that won't work here, because I can't see what I would do if that limit was reached. I can't just tell students to stop coding! What I want to do is calculate an upper bound on how much we will spend, not impose a limit on how much we can spend.

If it helps, these are the details of the module I am thinking of using GitHub classroom on:
* It has about 200 students. 
* It runs for  12 weeks and students are expected to spend 20 hours a week studying. More than half of the study time would be spent coding.
* I anticipate that students will code offline and upload their work to GitHub classroom to see the results of autograding tests. I do not plan to use Codespaces.

The question I am struggling with is how to convert those details into CI/CD minutes for the organization and CI/CD minutes for each student.
